homeassistant/components/todo/__init__.py

```diff
@@ -537,3 +543,16 @@ async def _async_remove_completed_items(entity: TodoListEntity, _: ServiceCall)
     ]
     if uids:
         await entity.async_delete_todo_items(uids=uids)
+
+
+async def _async_reset_todo_items(entity: TodoListEntity, _: ServiceCall) -> None:
+    """Reset all items from the To-do list to needed."""
+    items = [
+        item
+        for item in entity.todo_items or ()
+        if item.status != TodoItemStatus.NEEDS_ACTION
+    ]
+    for item in items:
+        data = dataclasses.asdict(item)
+        data["status"] = TodoItemStatus.NEEDS_ACTION
+        await entity.async_update_todo_item(item=TodoItem(**data))
```

Comment:
There's a potential race condition here, e.g. if something else deletes or renames the item before it's updated.